PowerValidatedSolutions.psm1

```diff
@@ -48145,6 +48145,383 @@ Function Test-WMSubnetInput {
+				if(Test-Path -Path $rootCaFilePathTemp) {
+					Remove-Item $rootCaFilePathTemp
+					break;
+				}
+			} else {
+				if($createRandomCounter -gt 6) {
+					Write-Error "Error: Unable to create temporary files in directory $certDir." -ErrorAction Stop
+					Exit
+				}
+			}
+		}
+        $rootCaFilename = "$caFqdn-rootCA.pem"
+        $rootCaFilePath = Join-Path -Path $certDir -ChildPath $rootCaFilename
+        if (Test-Path -Path $rootCaFilePath) {
+            Write-Error "Error: $rootCaFilePath file exists.  Please remove or rename the existing file." -ErrorAction Stop
+        }
+        if (-Not ([bool] (Get-Command -ErrorAction Ignore -Type Application openssl))) {
+            Write-Error "Error: Openssl not found.  Please verify openssl is installed."
+        }
+    }
+
+    # Create form array
+    $formFields = @{
+        Mode = "newreq"
+        FriendlyType = "Saved-Request Certificate"
+        CertRequest = Get-Content -Path $csrFilePath -Raw
+        CertAttrib = $CertAttributes
+        TargetStoreFlags = 0
+        SaveCert = "yes"
+    }
+
+    # Request CSR to be signed by MSCA
+    Try {
+        $CertSubmitted = Invoke-WebRequest -Uri "https://$caFqdn/certsrv/certfnsh.asp" -Credential $domainCreds -Method Post -Body $formFields -SkipCertificateCheck
+        $RequestID = ($CertSubmitted.content -split '\n' | Where-Object -FilterScript {$_ -match "certnew.cer\?ReqID=[0-9]"}) -replace '[^0-9]'
+        $null = Invoke-WebRequest -Uri "https://$caFqdn/certsrv/certnew.cer?ReqID=${RequestID}&Enc=b64" -Credential $domainCreds -OutFile $crtFilePath -SkipCertificateCheck
+        if ($getCArootCert.IsPresent) {
+            $null = Invoke-WebRequest -Uri "https://$caFqdn/certsrv/certnew.p7b?ReqID=CACert&Enc=b64" -Credential $domainCreds -OutFile $rootCaFilePathTemp -SkipCertificateCheck
+            openssl pkcs7 -inform PEM -outform PEM -in $rootCaFilePathTemp -print_certs > $rootCaFilePath
+        }
+    } catch {
+        Debug-ExceptionWriter -object $_
+    }
+
+    # Clean up
+    Remove-Item $rootCaFilePathTemp
+
+}
+Export-ModuleMember -Function Invoke-RequestSignedCertificateByMsca
+
+Function Invoke-GenerateChainPemFile {
+<#
+        .SYNOPSIS
+        Combine private key, signed certificate and/or root certificate chain into a single PEM file.   
+
+        .DESCRIPTION
+        The Invoke-GenerateChainPemFile cmdlet takes in private key, signed certificate and/or root certificate files and combine them into a single PEM file.  
+        
+        .EXAMPLE
+        Invoke-GenerateChainPemFile -certDir "C:\certificates" -keyFilePath "C:\certificates\xint-vrli01.rainpole.io.key" -crtFilePath 
+        "C:\certificates\xint-vrli01.rainpole.io.crt" -rootCaFilePath "C:\certificates\rpl-ad01.rainpole.io-rootCA.pem"
+        This example will combine xint-vrli01's private key, signed certificate and rpl0-ad01's root chain certificate into single
+        xint-vrli01.bundle.pem file. 
```

Comment:
```suggestion
        This example will combine sfo-vrli01's private key, signed certificate and rpl0-ad01's root chain certificate into single sfo-vrli01.bundle.pem file. 
```

Reply:
Addressed in commit 53039acbaaffafaf219274cac1a3684735130f53